src/server/visual-diff-plugin.js

```diff
@@ -0,0 +1,85 @@
+import { access, constants, stat } from 'node:fs/promises';
+import { dirname, join } from 'path';
+import { env } from 'node:process';
+
+const isCI = !!env['CI'];
+const ciDir = isCI ? 'ci' : '';
+const DEFAULT_MARGIN = 10;
+
+async function checkFileExists(fileName) {
+	try {
+		await access(fileName, constants.F_OK);
+		return true;
+	} catch (e) {
+		return false;
+	}
+}
+
+function extractTestPartsFromName(name) {
+	name = name.toLowerCase();
+	const parts = name.split(' ');
+	if (parts.length > 1) {
+		let dirName = parts.shift();
+		if (dirName.startsWith('d2l-')) {
+			dirName = dirName.substring(4);
+		}
+		return {
+			dir: dirName,
+			newName: parts.join('-')
+		};
+	}
+	return {
+		dir: '',
+		newName: parts.join('-')
+	};
+}
+
+export function visualDiff() {
+	return {
+		name: 'brightspace-visual-diff',
+		async executeCommand({ command, payload, session }) {
+
+			if (command !== 'brightspace-visual-diff') {
+				return;
+			}
+			if (session.browser.type !== 'playwright') {
+				throw new Error('Visual-diff is only supported for browser type Playwright');
+			}
+
+			const browser = session.browser.name.toLowerCase();
+			const { dir, newName } = extractTestPartsFromName(payload.name);
+			const goldenFileName = `${join(dirname(session.testFile), 'screenshots', ciDir, 'golden', browser, dir, newName)}.png`;
+			const currentFileName = `${join(dirname(session.testFile), 'screenshots', ciDir, 'current', browser, dir, newName)}.png`;
```

Comment:
Doesn't need to be figured out now, but based on the crazy names we've seen in the past hitting Windows limits, wondering if we want to make this shorter.  I'm not particularly attached to `screenshots`

Reply:
That's a good point -- could be `vdiff`?

Reply:
I'm good with being consistent everywhere with `vdiff`.

Reply:
Merging as-is for now, and will address this probably in the golden migration PR.